libraries/chain/include/graphene/chain/protocol/chain_parameters.hpp

```diff
@@ -67,6 +67,15 @@ namespace graphene { namespace chain {
 
       /** defined in fee_schedule.cpp */
       void validate()const;
+      
+      chain_parameters();
+      chain_parameters(const chain_parameters& other);
+      chain_parameters(chain_parameters&& other);
+      chain_parameters& operator=(const chain_parameters& other);
+      chain_parameters& operator=(chain_parameters&& other);
```

Comment:
Hm... the rule of 5 says that if you define these 4 then you must also define a destructor. You don't need the destructor because the `shared_ptr` takes care of that automatically. This opens the question, what does the `shared_ptr` *not* take care of automatically? Can't you get away with using default copy/move construct/assign?

The only thing you do need IMO is the no-arg constructor, to make sure the `shared_ptr` is initialized with an empty fee schedule.

Reply:
Right, thanks.